### ISSUE TITLE
Error when translate a point

### ISSUE THREAD

**Issue by johnnv1** [shapely/shapely#1485]
## Expected behavior and actual behavior.

Something in the last release (1.8.3) seems to cause an error when translate a point. Working fine with polygons.

## Steps to reproduce the problem.

```python
from shapely.geometry import Point
from shapely import affinity
affinity.translate(Point(10, 10), 50, 50)
```
```output
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/t/venv/lib/python3.8/site-packages/shapely/affinity.py", line 264, in translate
    return affine_transform(geom, matrix)
  File "/tmp/t/venv/lib/python3.8/site-packages/shapely/speedups/__init__.py", line 73, in affine_transform
    return _speedups.affine_transform(geom, matrix)
  File "shapely/speedups/_speedups.pyx", line 521, in shapely.speedups._speedups.affine_transform
  File "shapely/speedups/_speedups.pyx", line 553, in shapely.speedups._speedups.affine_transform
TypeError: an integer is required
```

## Env
Ubuntu 20.04
Shapely version: 1.8.3

**Comment by jorisvandenbossche** [comment]
@johnnv1 thanks for the report!

That's a regression in 1.8.3, I think caused by https://github.com/shapely/shapely/pull/1479 (cc @sgillies)

It's a bit strange that we didn't catch this in our CI

**Comment by jorisvandenbossche** [comment]
> It's a bit strange that we didn't catch this in our CI

It seems that whether it errors or not depends on how the geometry was creates (and our affinity tests use WKT construction, which doesn't trigger the error, and thus this wasn't caught in our CI):

```
In [1]: import shapely

In [2]: shapely.__version__
Out[2]: '1.8.3'

In [3]: from shapely.geometry import Point
   ...: from shapely import affinity
   ...: from shapely import wkt

In [4]: p = Point(1, 1)

In [5]: affinity.translate(p)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [5], in <cell line: 1>()
----> 1 affinity.translate(p)

File ~/miniconda3/envs/shapely-183/lib/python3.10/site-packages/shapely/affinity.py:264, in translate(geom, xoff, yoff, zoff)
    259     return geom
    260 matrix = (1.0, 0.0, 0.0,
    261           0.0, 1.0, 0.0,
    262           0.0, 0.0, 1.0,
    263           xoff, yoff, zoff)
--> 264 return affine_transform(geom, matrix)

File ~/miniconda3/envs/shapely-183/lib/python3.10/site-packages/shapely/speedups/__init__.py:73, in enable.<locals>.affine_transform(geom, matrix)
     72 def affine_transform(geom, matrix):
---> 73     return _speedups.affine_transform(geom, matrix)

File ~/miniconda3/envs/shapely-183/lib/python3.10/site-packages/shapely/speedups/_speedups.pyx:521, in shapely.speedups._speedups.affine_transform()

File ~/miniconda3/envs/shapely-183/lib/python3.10/site-packages/shapely/speedups/_speedups.pyx:553, in shapely.speedups._speedups.affine_transform()

TypeError: an integer is required

In [6]: p = wkt.loads("POINT (1 1)")

In [7]: affinity.translate(p)
Out[7]: <shapely.geometry.point.Point at 0x7f25d4bb87c0>
```

**Comment by sgillies** [comment]
Shapely releases are pretty cheap to do, I could publish a 1.8.4 today (https://github.com/shapely/shapely/issues/1486).

**Comment by jorisvandenbossche** [comment]
So the underlying reason here is that `_geom` / `__geom__` no longer is an int:

```
In [3]: Point(10, 10).__geom__
Out[3]: <shapely.ctypes_declarations.LP_GEOSGeom_t at 0x7fbcf08ce5c0>
```

We are seeing similar problems in geopandas (with conversion to pygeos): https://github.com/geopandas/geopandas/issues/2531. So ideally we would solve this by changing `__geom__` back to an int, and not by updating our internal code to deal with the changed return value.

**Comment by jorisvandenbossche** [comment]
One quick option would be to revert https://github.com/shapely/shapely/pull/1479 (since it also seems to not have solved https://github.com/shapely/shapely/issues/1449, which originally triggered that fix)

Otherwise https://github.com/shapely/shapely/pull/1488 could be a potential fix to keep the return value of `__geom__`

**Comment by jorisvandenbossche** [comment]
Ah, I see you already just opened a PR with a revert. So https://github.com/shapely/shapely/pull/1488 could also be an option, either way is fine for me.

**Comment by sgillies** [comment]
Resolved by #1487.
